src/Algebra/Graph/Relation/Symmetric.hs

```diff
@@ -11,27 +11,485 @@
 -----------------------------------------------------------------------------
 module Algebra.Graph.Relation.Symmetric (
     -- * Data structure
-    SymmetricRelation, fromRelation, toRelation,
+    SymmetricRelation(..), fromRelation, toRelation,
+
+    -- * Basic graph construction primitives
+    empty, vertex, edge, overlay, connect, vertices, edges, overlays, connects,
+
+    -- * Relations on graphs
+    isSubgraphOf,
 
     -- * Graph properties
-    neighbours
+    isEmpty, hasVertex, hasEdge, vertexCount, edgeCount, vertexList, edgeList,
+    adjacencyList, vertexSet, edgeSet, neighbours,
+
+    -- * Standard families of graphs
+    path, circuit, clique, biclique, star, stars, tree, forest,
+
+    -- * Graph transformation
+    removeVertex, removeEdge, replaceVertex, mergeVertices, gmap, induce,
+
+    -- * Relational operations
+    compose
   ) where
 
-import Algebra.Graph.Relation
-import Algebra.Graph.Relation.InternalDerived
+import qualified Algebra.Graph.Relation as R
+import qualified Algebra.Graph.Relation.Internal as RI hiding (empty, vertex, overlay, connect, referredToVertexSet)
+import Algebra.Graph.Relation.Symmetric.Internal
 
 import qualified Data.Set as Set
+import Data.Tree 
+import qualified Data.Tree as Tree
+import Data.Tuple (swap)
 
 -- | Construct a symmetric relation from a 'Relation'.
--- Complexity: /O(1)/ time.
-fromRelation :: Relation a -> SymmetricRelation a
-fromRelation = SymmetricRelation
+-- Complexity: /O(m*log(m))/ time.
+--
+-- @
+-- fromRelation ('Algebra.Graph.Relation.edge' 1 2) == 'edge 1 2'
+-- fromRelation . 'toRelation'                      == id
+-- 'vertexCount' . fromRelation                     == 'Algebra.Graph.Relation.vertexCount'
+-- (* 2) . 'edgeCount' . fromRelation               >= 'Algebra.Graph.Relation.edgeCount'
+-- @
+fromRelation :: Ord a => R.Relation a -> SymmetricRelation a
+fromRelation = SymmetricRelation . R.symmetricClosure
 
 -- | Extract the underlying relation.
--- Complexity: /O(m*log(m))/ time.
-toRelation :: Ord a => SymmetricRelation a -> Relation a
-toRelation = symmetricClosure . fromSymmetric
+-- Complexity: /O(1)/ time.
+--
+-- @
+-- toRelation ('edge' 1 2)                           == 'Algebra.Graph.Relation.edges' [(1,2), (2,1)]
+-- 'Algebra.Graph.Relation.vertexCount' . toRelation == 'vertexCount'
+-- 'Algebra.Graph.Relation.edgeCount' . toRelation   <= (* 2) . 'edgeCount'
+-- @
+toRelation :: SymmetricRelation a -> R.Relation a
+toRelation = fromSymmetric
```

Comment:
Since we are going to rename `SymmetricRelation` to simply `Relation`, this name will become a bit confusing. I suggest to rename this function to `fromSymmetric`, just as it's named already :)